libs/core/src/lib/components/wrapper-component.ts

```diff
@@ -219,17 +218,17 @@ export abstract class ReactWrapperComponent<TProps extends {}> implements AfterV
       return (context: TContext) => htmlRenderer.render(context);
     }
 
-    if (input instanceof RenderOverride) {
-      return undefined;
-    }
-
     if (typeof input === 'object') {
-      const { componentType, factoryResolver, injector } = input;
-      const componentFactory = factoryResolver.resolveComponentFactory(componentType);
-      const componentRef = componentFactory.create(injector);
-
-      // Call the function again with the created ComponentRef<TContext>
-      return this.createInputJsxRenderer(componentRef, additionalProps);
+      if ('componentType' in input) {
```

Comment:
please replace this with a [user defined type guard](https://basarat.gitbooks.io/typescript/docs/types/typeGuard.html#user-defined-type-guards), this function does too much at the moment (it was bearable so far since it only had to deal with one `object`, but in hindsight it should've extracted the "is this a `RenderComponentOptions`" to a user defined type guard.

Also, you want to check with `hasOwnProperty`, since it doesn't look up the prototype chain.